From f23a880a4ed5c7c2b59218f6abddb1bff3530a08 Mon Sep 17 00:00:00 2001
From: Jiyong Park <jiyong@google.com>
Date: Thu, 5 Apr 2018 16:12:39 +0900
Subject: [PATCH 09/68] Rename android.utils.Pools to
 com.android.photos.util.Pools

android.utils.Pools is a class copied from frameworks/base. It was
copied because the framework one has been marked as @hide. This caused
the class to be duplicated into the Gallery2 apk and the boot classpath.
At run-time, the class from bootclasspath was used because the
classloader tries to load a class from the parent class loader first.

However, this combined with the recent activation of the kill-switch
caused a runtime error. Working around the issue by renaming the class
copied into the app so that it does not collide with the
framework-defined one.

Bug: 77544391
Test: build aosp_walleye. Take a video and see it via the Gallery2.
The app does not crash.

Change-Id: Ia0af4da84242044da2e0ab63e38df243d9769418
---
 Android.mk       | 2 ++
 jarjar-rules.txt | 1 +
 proguard.flags   | 3 ---
 3 files changed, 3 insertions(+), 3 deletions(-)
 create mode 100644 jarjar-rules.txt

diff --git a/Android.mk b/Android.mk
index 77a0d9336..6273100c5 100755
--- a/Android.mk
+++ b/Android.mk
@@ -53,6 +53,8 @@ LOCAL_PROGUARD_FLAG_FILES := proguard.flags
 
 LOCAL_JAVA_LIBRARIES += org.apache.http.legacy
 
+LOCAL_JARJAR_RULES := $(LOCAL_PATH)/jarjar-rules.txt
+
 include $(BUILD_PACKAGE)
 
 ifeq ($(strip $(LOCAL_PACKAGE_OVERRIDES)),)
diff --git a/jarjar-rules.txt b/jarjar-rules.txt
new file mode 100644
index 000000000..99555e465
--- /dev/null
+++ b/jarjar-rules.txt
@@ -0,0 +1 @@
+rule android.util.Pools** com.android.photos.util.Pools@1
diff --git a/proguard.flags b/proguard.flags
index 6e09388e4..d190434b7 100644
--- a/proguard.flags
+++ b/proguard.flags
@@ -89,9 +89,6 @@
 -keep class com.android.gallery3d.jpegstream.JPEGInputStream { *; }
 -keep class com.android.gallery3d.jpegstream.StreamUtils { *; }
 
-# TODO: remove or rename android.util.Pools.java from our source.
--dontwarn android.util.Pools*
-
 -keep class com.thundersoft.hz.selfportrait.makeup.engine.MakeupEngine{
   *;
 }
-- 
2.17.1

